extensions/json-jackson/src/main/java/io/deephaven/json/jackson/Parsing.java

```diff
@@ -0,0 +1,303 @@
+        if (parser.hasCurrentToken()) {
+            throw new IllegalStateException(
+                    String.format("Expected no current token. actual=%s", parser.currentToken()));
+        }
+    }
+
+    static void assertNextToken(JsonParser parser, JsonToken expected) throws IOException {
+        final JsonToken actual = parser.nextToken();
+        if (actual != expected) {
+            throw new IllegalStateException(
+                    String.format("Unexpected next token. expected=%s, actual=%s", expected, actual));
+        }
+    }
+
+    static void assertCurrentToken(JsonParser parser, JsonToken expected) {
+        if (!parser.hasToken(expected)) {
+            throw new IllegalStateException(
+                    String.format("Unexpected current token. expected=%s, actual=%s", expected, parser.currentToken()));
+        }
+    }
+
+    static CharSequence textAsCharSequence(JsonParser parser) throws IOException {
+        return parser.hasTextCharacters()
+                ? CharBuffer.wrap(parser.getTextCharacters(), parser.getTextOffset(), parser.getTextLength())
+                : parser.getText();
+    }
+
+    static class UnexpectedToken extends JsonProcessingException {
+        public UnexpectedToken(String msg, JsonLocation loc) {
+            super(msg, loc);
+        }
+    }
+
+    static IOException mismatch(JsonParser parser, Class<?> clazz) {
+        final JsonLocation location = parser.currentLocation();
+        final String msg = String.format("Unexpected token '%s'", parser.currentToken());
+        return new UnexpectedToken(msg, location);
+    }
+
+    static IOException mismatchMissing(JsonParser parser, Class<?> clazz) {
+        final JsonLocation location = parser.currentLocation();
+        return new UnexpectedToken("Unexpected missing token", location);
+    }
+
+    static byte parseStringAsByteBool(JsonParser parser, byte onNull) throws IOException {
+        final String text = parser.getText().trim();
+        if ("true".equalsIgnoreCase(text)) {
+            return BooleanUtils.TRUE_BOOLEAN_AS_BYTE;
+        }
+        if ("false".equalsIgnoreCase(text)) {
+            return BooleanUtils.FALSE_BOOLEAN_AS_BYTE;
+        }
+        if ("null".equalsIgnoreCase(text)) {
+            return onNull;
+        }
+        throw new IOException(String.format("Unexpected string as boolean '%s'", text));
+    }
+
+    static Boolean parseStringAsBoolean(JsonParser parser, Boolean onNull) throws IOException {
+        final String text = parser.getText().trim();
+        if ("true".equalsIgnoreCase(text)) {
+            return Boolean.TRUE;
+        }
+        if ("false".equalsIgnoreCase(text)) {
+            return Boolean.FALSE;
+        }
+        if ("null".equalsIgnoreCase(text)) {
+            return onNull;
+        }
+        throw new IOException(String.format("Unexpected string as boolean '%s'", text));
+    }
+
+    static char parseStringAsChar(JsonParser parser) throws IOException {
```

Comment:
What happens when users do awful things with characters that don't fit in a single `char`? I think this just needs a test case.

Reply:
There will be an exception that string length needs to be 1.